openfe/protocols/openmm_septop/base.py

```diff
@@ -0,0 +1,1436 @@
+        sampler_context_cache : openmmtool.cache.ContextCache
+          Context cache for the sampler states.
+
+        Returns
+        -------
+        sampler : multistate.MultistateSampler
+          A sampler configured for the chosen sampling method.
+        """
+        rta_its, rta_min_its = settings_validation.convert_real_time_analysis_iterations(
+            simulation_settings=simulation_settings,
+        )
+        et_target_err = settings_validation.convert_target_error_from_kcal_per_mole_to_kT(
+            thermo_settings.temperature,
+            simulation_settings.early_termination_target_error,
+        )
+
+        # Select the right sampler
+        # Note: doesn't need else, settings already validates choices
+        if simulation_settings.sampler_method.lower() == "repex":
+            sampler = multistate.ReplicaExchangeSampler(
+                mcmc_moves=integrator,
+                online_analysis_interval=rta_its,
+                online_analysis_target_error=et_target_err,
+                online_analysis_minimum_iterations=rta_min_its
+            )
+        elif simulation_settings.sampler_method.lower() == "sams":
+            sampler = multistate.SAMSSampler(
+                mcmc_moves=integrator,
+                online_analysis_interval=rta_its,
+                online_analysis_minimum_iterations=rta_min_its,
+                flatness_criteria=simulation_settings.sams_flatness_criteria,
+                gamma0=simulation_settings.sams_gamma0,
+            )
+        elif simulation_settings.sampler_method.lower() == 'independent':
+            sampler = multistate.MultiStateSampler(
+                mcmc_moves=integrator,
+                online_analysis_interval=rta_its,
+                online_analysis_target_error=et_target_err,
+                online_analysis_minimum_iterations=rta_min_its,
+            )
+
+        sampler.create(
+            thermodynamic_states=cmp_states,
+            sampler_states=sampler_states,
+            storage=reporter
+        )
+
+        sampler.energy_context_cache = energy_context_cache
+        sampler.sampler_context_cache = sampler_context_cache
+
+        return sampler
+
+    def _run_simulation(
+            self,
+            sampler: multistate.MultiStateSampler,
+            reporter: multistate.MultiStateReporter,
+            settings: dict[str, SettingsBaseModel],
+            dry: bool,
+            verbose: bool,
+            shared_basepath: pathlib.Path,
+    ):
+        """
+        Run the simulation.
+
+        Parameters
+        ----------
+        sampler : multistate.MultiStateSampler
+          The sampler associated with the simulation to run.
+        reporter : multistate.MultiStateReporter
+          The reporter associated with the sampler.
+        settings : dict[str, SettingsBaseModel]
+          The dictionary of settings for the protocol.
+        dry : bool
+          Whether or not to dry run the simulation
```

Comment:
If you do need to pass in verbose / shared_basepath, then these should be reflected in the docstring.